Doc/whatsnew/3.12.rst

```diff
@@ -382,6 +390,10 @@ Changes in the Python API
   to :term:`filesystem encoding and error handler`.
   Argument files should be encoded in UTF-8 instead of ANSI Codepage on Windows.
 
+* Removed the strictly synchronous ``smtpd`` module, deprecated in Python 3.4.7
```

Comment:
Technically `smtpd` isn't synchronous, it's just built on the deprecated `asynchat`/`asyncore` framework.